src/components/site/header/HeaderProgress.jsx

```diff
@@ -0,0 +1,49 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import CustomIcon from '../icons/CustomIcon';
+import LinearProgress from '../linearProgress';
+import LinearProgressMarker from '../linearProgress/LinearProgressMarker';
+
+class HeaderProgress extends React.PureComponent {
+  render = () => {
+    const { checkpoints, pageNumber, totalPages } = this.props;
+    return (
+      <div className="header-progress-wrapper">
+        <LinearProgress
+          id="current-page-of-total"
+          value={pageNumber ? Math.round((pageNumber / totalPages) * 100) : 0}
+        >
+          {checkpoints.length > 0 && (
```

Comment:
add the checkpoints into the header's linear progress bar if there are any. Add the finish marker.